### ISSUE TITLE
[DPI] Elements required works to properly support hiDPI

### ISSUE THREAD

**Issue by ozone10** [notepad-plus-plus/notepad-plus-plus#14959]
This issue is for tracking issues with hiDPI environment and works on it.

## Main window
 - [ ] Tab control
 - [ ] Toolbar
 - [ ] Editor
 - [ ] Status bar

## Dialogs
- [ ] FindInFinder
- [ ] About
- [ ] DebugInfo
- [ ] Run 
- [ ] HashFromFiles
- [ ] GoToLine
- [ ] ColumnEditor 
- [ ] Style Configurator 
- [ ] Preference
- [ ] FindCharsInRange 
- [ ] Rename Current Tab
- [ ] Save all
- [ ] Shortcut
- [ ] Run Macro Multiple Times
- [ ] UDL styler

## More complex
- [ ] Find and Replace
- [ ] PluginsAdmin 
- [ ] Windows 
- [ ] Shortcut Mapper
- [ ] Progress
- [ ] Color Popup
- [ ] Color dialog
- [ ] UDL

## Panels
- [ ] Title
- [ ] Tab control
- [ ] Main field
- [ ] Plugin support

**Comment by donho** [comment]
@ozone10 

Just found 2 dialogs (Preferences & Find Replace) partially working & 1 (Shortcut mapper) totally not working:
  
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/425d4a06-f663-4cad-abe1-e287aa4a8059)

**Comment by ozone10** [comment]
@donho 
I will be away for 2 weeks.

For preference add `DPIManagerV2::setDpiWithParent(_hSelf);` before line
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/018d4f9a1f01a4fc17ebc1d0d8e372db1cf5539d/PowerEditor/src/WinControls/Preference/preferenceDlg.cpp#L1541

Shortcut mapper, Find&Replace and Windows dialogs are sizeable which require more works.

btw
While working on toolbar I noticed some issue with custom icons for toolbar
Could you check it:
1. Have custom icons for toolbar enabled.
2. Open preference and change between large and small icons several times.

**Comment by donho** [comment]
> Just found 2 dialogs (Preferences & Find Replace) partially working & 1 (Shortcut mapper) totally not working:

@ozone10 
Sorry, I wasn't clear. What I meant is:
After changing `<dpiAwareness>system, unaware</dpiAwareness>` to `<dpiAwareness>PerMonitorV2, system, unaware</dpiAwareness>`, 2 dialogs (Preferences & Find Replace) partially working & 1 (Shortcut mapper) totally not working under a multi-monitor environment. With `<dpiAwareness>system, unaware</dpiAwareness>` under the multi-monitor environment, Notepad++ is just blurry.

> Replacing old NppParameters::getInstance()._dpiManager.scaleX and Y variant with DPIManagerV2::scale will help with initial looks.

So it's not necessary I guess. All I wanted to point out is, if `<dpiAwareness>PerMonitorV2, system, unaware</dpiAwareness>` is used, there are some issue in these 3 dialogs - and it's needed to rework in PerMonitorV2 mode.

**Comment by donho** [comment]
BTW, the provided screenshot is Notepad++ dragged from a scale 225% laptop screen to a scale 100% external monitor.

**Comment by ozone10** [comment]
@donho 
> Can you reproduce what I've shown you in the screenshot with `<dpiAwareness>PerMonitorV2, system, unaware</dpiAwareness>`? If yes, that's a good news, all we need is fix the bugs.

Yes that is right, because they are using dpi scaling from main monitor.

For preferences dialog for some reason subdialogs when initializing will use dpi scaling from main monitor or system wide. 
So adding `DPIManagerV2::setDpiWithParent(_hSelf);`  to subdialogs or `DPIManagerV2::setDpi(_hSelf);` in main dialog in `WM_INITDIALOG` will fix it.

**Comment by ozone10** [comment]
BTW
it is possible to have mixed mode, some dialogs PerMonitorV2 aware and some only system aware.

**Comment by donho** [comment]
> it is possible to have mixed mode, some dialogs PerMonitorV2 aware and some only system aware.

How can it be possible?

> For preferences dialog for some reason subdialogs when initializing will use dpi scaling from main monitor or system wide.
So adding DPIManagerV2::setDpiWithParent(_hSelf); to subdialogs or DPIManagerV2::setDpi(_hSelf); in main dialog in WM_INITDIALOG will fix it.

There is only Dark mode dialog is not working in preferences dialog. The others work fine.

**Comment by donho** [comment]
> There is only Dark mode dialog is not working in preferences dialog. The others work fine.

And Dark mode dialog doesn't work fine when Notepad++ is launched on the external monitor (scale 100), otherwise it works fine. And the provided screenshot is while Notepad++ being launched on the external monitor (scale 100) - indeed it's because they are using dpi scaling from main monitor, as you have said.

Shortcut mapper and Find replace have more problem. I notice that if Notepad++ is launched on my laptop, open both Shortcut mapper and Find replace on scale 225% laptop screen are OK. However, when I drag them to the external monitor (scale 100), both resize to larger.

As well, when I drag the Find replace back to laptop screen (scale 225%), and stay a little more between 2 screens, the dialog blinks crazily,  and the controls inside turn out hiding toward left topper side. I believe it's due to the nature of Find replace dialog which is resizable.

**Comment by donho** [comment]
> So adding DPIManagerV2::setDpiWithParent(_hSelf); to subdialogs or DPIManagerV2::setDpi(_hSelf); in main dialog in WM_INITDIALOG will fix it.

Indeed, the commit https://github.com/notepad-plus-plus/notepad-plus-plus/commit/3c06d89b2ae837b74bff39390130251bf7fa54f4 fixed it.

For wrong display of Shortcut mapper, I guess WM_DPICHANGED is missing in BabyGrid.cpp

**Comment by ozone10** [comment]
@donho 
> > it is possible to have mixed mode, some dialogs PerMonitorV2 aware and some only system aware.
> 
> How can it be possible?

code from #15029

```c++
void ColourPopup::createForDpi(int dialogID)
{
	const DPI_AWARENESS_CONTEXT dpiContextSystem = NppDarkMode::isWindows10() ? DPI_AWARENESS_CONTEXT_UNAWARE_GDISCALED : DPI_AWARENESS_CONTEXT_SYSTEM_AWARE;
	const auto dpiContext = DPIManagerV2::setThreadDpiAwarenessContext(dpiContextSystem);
	create(dialogID);
	if (dpiContext != NULL)
	{
		DPIManagerV2::setThreadDpiAwarenessContext(dpiContext);
	}
}
```
`setThreadDpiAwarenessContext` will allow to set dpi mode per top window

Color dialog and Color popup have many controls that are not automatically resizable by PerMonitorV2 dpi mode and would require a lot of coding to properly support it.
So in PR #15029 I use `DPI_AWARENESS_CONTEXT_UNAWARE_GDISCALED` ("System (Enhanced)") if supported which is good enough.

**Comment by ozone10** [comment]
@donho 
PR #15029 will be last from me for while. I will be away. 
Could you also check issue with toolbar https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14959#issuecomment-2068133497, with this bug it's impossible to make dpi working with toolbar.

**Comment by donho** [comment]
@ozone10 
> btw
While working on toolbar I noticed some issue with custom icons for toolbar
Could you check it:

>    Have custom icons for toolbar enabled.

What do you mean by "custom icons"?

>    Open preference and change between large and small icons several times.

It (changing between large and small icons several times) doesn't change the size of tool icons - they are still big and not proportional with other elements (menu bar and Scintilla edit zone).

**Comment by donho** [comment]
@ozone10 
Regarding toolbar (tool icon issue), I have tested in a duel screen laptop: the main display has been reset to scale 150%, and the 2nd display scaled to 200%. The result is toolbar size is always 150% in both displays, and the menu size resizes correctly (150% <-> 200%).
So I suppose toolbar's initializing uses dpi scaling from main monitor, as dark mode of preferences dialog.

**Comment by ozone10** [comment]
@donho
> Regarding toolbar (tool icon issue), I have tested in a duel screen laptop: the main display has been reset to scale 150%, and the 2nd display scaled to 200%. The result is toolbar size is always 150% in both displays, and the menu size resizes correctly (150% <-> 200%).
> So I suppose toolbar's initializing uses dpi scaling from main monitor, as dark mode of preferences dialog.

Yes that is correct.

> What do you mean by "custom icons"?

https://npp-user-manual.org/docs/config-files/#toolbar-icon-customization-toolbariconsxml


![toolbar](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/55940305/f44950fa-0d09-4a3c-b4f8-dd15636edc0f)

**Comment by donho** [comment]
@ozone10 
> Have custom icons for toolbar enabled.

No, as you can see in the [screenshot ](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14959#issuecomment-2068108871)I've provided. The custom icons are not enabled.

**Comment by ozone10** [comment]
@donho 
I think there is misunderstanding.
Handling dpi changes when moving between two screens will require reloading icons. For in-built ones there is no issue I even had PR for it, but during testing I've found out that there is bug with custom icons.

As you can see in gif above this bug can be triggered in current v.8.6.5, because changing between small and large icon set will reload icons and if custom icons are used there will be leak.

**Comment by donho** [comment]
@ozone10 
> As you can see in gif above this bug can be triggered in current v.8.6.5, because changing between small and large icon set will reload icons and if custom icons are used there will be leak.

Yes, I can see it in the gif. It's surely a bug. However, the custom icons feature is used by few people, I think 1stly we should focus on the toolbar with the builtin icons - to make it resize correctly among the screens.

**Comment by donho** [comment]
> edit 3 files to have basic dpi support for toolbar...

@ozone10 
Done:
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15044

Tested, but the problem resists. :(

**Comment by ozone10** [comment]
@donho 
> Tested, but the problem resists. :(

well, you did not replace every `NppParameters::getInstance()._dpiManager.scaleX` with `DPIManagerV2::scale`

**Comment by donho** [comment]
@ozone10 
> well, you did not replace every NppParameters::getInstance()._dpiManager.scaleX with DPIManagerV2::scale

It works now: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15044/commits/b97afee33a5b5915485fdfc4605795c9162485f1

Thank you!

**Comment by donho** [comment]
@ozone10 
FYI: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15044/commits/fd014859a01e5f758448ef943628df59f89f0543

**Comment by ozone10** [comment]
@donho 
FindInFinder should already work
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/55940305/cf8ffb11-feb3-4dfa-9e61-9c1a6def0a44)

**Comment by WildByDesign** [comment]
@ozone10 @donho I brought the Fluent UI icons over the npp a few years back. I was wondering if there is anything that we can do regarding icons that can make the situation better for higher DPI.

For example, when I did the original work on the Fluent UI icons for npp, I was using 100% scaling at the time. Now I use 125% scaling the majority of the time and all of the 16px icons get scaled to 20px and therefore lose some sharpness.

I can't imagine adding another potentially 100+ icons as 20px, plus 100+ icons at other resolutions.

Some possibilities:

- Is there any possibility to switch to SVG for the toolbar icons?

- Another possibility is that, very recently, NanaZip added support (https://github.com/M2Team/NanaZip/commit/4e70eb18d135ea29255da6aa8234c2c64b20a87b) where they switched to XAML and using fonts (specifically, Segoe MDL2 Assets) for the toolbar icons.

Anyway, I might be willing and able to do some icon work but I just don't know what the best direction to go is. I am not a programmer so I can't help with coding, unfortunately.

**Comment by donho** [comment]
@WildByDesign 
Without Fluent UI icons, Notepad++ dark mode wouldn't look so good  - Thank you!


Current releases are not compiled with the activation of DPI scaled by application. 
The following 2 screenshots is Notepad++ toolbar's Fluent UI icons under 350% scaled windows11.
The left image scaled by system (v8.6.7 release), the right one scaled by application (my local build):

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/f53b57cc-1a02-4ca6-919c-bd28c2e48921)

So as you can see, the right one looks better. 

> Is there any possibility to switch to SVG for the toolbar icons?

AFAIK, win32 API use only *.ico or *.bmp for the toolbar icons. But @ozone10 knows better win32 than I do.
@ozone10 is it possible? If it is, is it easy to switch to *.svg?

> Another possibility is that, very recently, NanaZip added support (https://github.com/M2Team/NanaZip/commit/4e70eb18d135ea29255da6aa8234c2c64b20a87b) where they switched to XAML and using fonts (specifically, Segoe MDL2 Assets) for the toolbar icons.

It could be considered, but switching to XAML in short or middle term is unlikely.

Let's see the future version of Notepad++ as multi-dpi awareness by application enabled. If the issue persists, we will seek the solution.

**Comment by ozone10** [comment]
@WildByDesign @donho 
using SVG or WinUI 3 ("XAML") would require major refactoring, which currently IMHO is not possible. WinUI 3 would be ideal, but it's "quite" new and many things are not properly documented.
For SVG we could use Direct2D, but currently the simplest direction is using multisize ico files:

for toolbar icons 16x16, 20x20, 24x24, 32x32, 40x40, 48x48, 64x64 and 96x96, 
for tab bar and toolbars from docking panels 16x16, 20x20, 24x24, 32x32 and 48x48, 
for docking panels tab bar 14x14, 16x16, 20x20, 24x24, 32x32 and 48x48

These sizes should be enough for up 300 % dpi scale.

**Comment by WildByDesign** [comment]
> The left image scaled by system (v8.6.7 release), the right one scaled by application (my local build):

I do my own local builds as well. What do I need to change to allow scaled by application in my own local build?

Thank you both for the thorough answers, I appreciate it. I’m going to go over the details in both comments and respond more later.

**Comment by ozone10** [comment]
@WildByDesign 
check test PR #13533 which has dpi scaling enabled (file [dpiAware.manifest](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/13533/files#diff-5acdbb8328ec6234d363ac589eff2634d41f8ad9bc6ce8b7a98cde64cff5ec0f)).

**Comment by WildByDesign** [comment]
I did some testing lastnight, checked my previous SVG sources, created my plan (local so far) and I am happy with my results at the moment.

![npp-toolbar-dpi](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/26308319/58468ee9-433a-4229-ae66-99d35e9eb94e)

This is only with the New icon done to show in comparison to the Open icon beside it which is not done. It's quite a difference and perfectly on the pixel lines which is crucial for icon sharpness.

I don't want to take away from this issue. So I will keep working away at the toolbar icons first. When I am closer to being ready, I will create a separate issue for the Icons and DPI Sizing followed by a PR at some point. I don't want to give a timeline yet because this is a massive amount of icons. But I am excited about it so far.

**Comment by donho** [comment]
> for toolbar icons 16x16, 20x20, 24x24, 32x32, 40x40, 48x48, 64x64 and 96x96, for tab bar and toolbars from docking panels 16x16, 20x20, 24x24, 32x32 and 48x48, for docking panels tab bar 14x14, 16x16, 20x20, 24x24, 32x32 and 48x48
> 
> These sizes should be enough for up 300 % dpi scale.

It'll be 256 icons only for the toolbar!
I wonder if that'll add size of binary (notepad++.exe) considerably.
Also it will add the logic of scale rate to specific size of icon in the new DPI management.

@ozone10  Could you confirm that?

**Comment by ozone10** [comment]
@donho 
ICO file can contain multiple sizes of images. Size of ico file depends on how are each images stored, usually they are using BMP format, but we want PNG which is smaller.
Scaling logic is already added
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/00ea1a90898cb1331276d757f537f47c1b562fba/PowerEditor/src/dpiManagerV2.cpp#L285

Here is example with souce svg and ico with 16x16, 20x20, 24x24, 32x32, 40x40, 48x48, 64x64 and 96x96 sizes
[icon.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/15421067/icon.zip)

**Comment by donho** [comment]
@ozone10 
You're right. Current toolbar icons are in ICO format. Thank you for the confirmation.

**Comment by WildByDesign** [comment]
> when creating ico files make sure that containing images are stored in png format.

Indeed, yes. I am doing all PNG (stored in ICO) of the pixel sizes that you mentioned a few posts back. I am approx. 40% done the toolbar buttons. They look so sharp and nice. I wish I had done it years ago.

I am not sure what we are going to do with all of the BMP icons though because there is no easy way to scale those. Is there any way to do ICO with those also?

**Comment by ozone10** [comment]
@WildByDesign 
if you are done with svg files, generating ico files is then easy, just use script to generate them. 

@donho 
it is possible to reorganize https://github.com/notepad-plus-plus/notepad-plus-plus/tree/master/PowerEditor/src/icons folder.
Make it same for standard and light fluent variant icons as with dark mode.

**Comment by donho** [comment]
> it is possible to reorganize https://github.com/notepad-plus-plus/notepad-plus-plus/tree/master/PowerEditor/src/icons folder.
Make it same for standard and light fluent variant icons as with dark mode.

OK.

**Comment by WildByDesign** [comment]
@ozone10 I made a script for converting the SVGs to PNG in all necessary DPI using Inkscape CLI for best quality. Then I created a second script that stuffs all of the PNGs of various DPI into ICO as the last step.

The part that is slowing me down is that a bunch of the icons never existed and I had to create the SVGs from scratch for some of them, following the same Fluent design guidelines.

Also some of the Fluent SVGs are missing some DPI sizes and I have had to create some of those because  SVGs don’t always scale as nicely as people often think. They absolutely must follow along the pixel lines to be sharp icons. There needs to be 3 base (16, 20, and 24) which can then be multiplied properly into all sizes that we need.

I am approx. 60% done now.

Do you guys want me to do a PR when the toolbar icons are complete or do you want me to send them before so that you can test them privately?

**Comment by ozone10** [comment]
@donho 
> No problem.

#15192

@WildByDesign 
> I am approx. 60% done now.

cool.

Also check PR #15192 for new icons folder structure.

> Do you guys want me to do a PR when the toolbar icons are complete or do you want me to send them before so that you can test them privately?

It depends on @donho, but IMHO I think you can do PR, we can test it from there.

**Comment by donho** [comment]
@WildByDesign 
> I am approx. 60% done now.

Sounds great.


> Do you guys want me to do a PR when the toolbar icons are complete or do you want me to send them before so that you can test them privately?

I prefer with PR, for the sake of that the merged PR gives you the credit of the work you've done directly, thanks to Git's authoring system. So people know who did contribution on which part directly from the Git master history (https://github.com/notepad-plus-plus/notepad-plus-plus/commits/master/). Personally, I think it's a fair system/way.

However, if you are not familiar with git/github, or you want to be an anonymous contributor, you can always attach the zip of icons here, I will integrate the icons for you.

**Comment by WildByDesign** [comment]
@donho Thank you, I appreciate it. I am only 1 custom icon away from the toolbar icons from being complete.

I just have to pull in the latest commit and adapt to the new icon directories layout, which looks much better by the way. Then I should be ready to do the PR later today.

I have tested the updated toolbar icons thoroughly in dark mode, but I will need more help testing them in light mode as well once the PR is done.

**Comment by WildByDesign** [comment]
I've just submitted the PR's (one for each directory). I couldn't figure out how to upload it all as one PR through Github.com web site.

**Comment by ozone10** [comment]
@WildByDesign 
unfortunately, those ico files contains bmp images. Generated ico files should be only around 10 kb large

Also if you are doing PR through web you can do multiple commit in one PR. But still try to learn basic git, it is very handy.

**Comment by WildByDesign** [comment]
@ozone10 Thank you for catching that. Please hold off on the PRs and I will recreate them properly later. Sorry about that.

**Comment by ozone10** [comment]
are you using ImageMagick convert?

**Comment by WildByDesign** [comment]
> are you using ImageMagick convert?

Yes. I have all of the PNGs already converted from SVG and simply have a script for ImageMagick to convert the PNG to ICO. But somehow that is converting the PNG to bmp. So I’m just doing some research on possibly appending `png8` to the script.

**Comment by ozone10** [comment]
Unfortunately ImageMagick convert does not support creating ico with png images with sizes smaller than 256x256.

I would use python with cairosvg library to make ico files from svg.

**Comment by WildByDesign** [comment]
Ok thank you. I will check it out.

**Comment by ozone10** [comment]
@WildByDesign 
could you provide some svg files for testing?

**Comment by WildByDesign** [comment]
@ozone10 Absolutely. 
[Replace (mine).zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/15461344/Replace.mine.zip)
Here is the logic that I have been following for the SVG to PNG conversion. If you need anymore, please let me know.

**Comment by WildByDesign** [comment]
It took me a few hours, but I have a solution now that adds the PNGs to ICO as proper PNG layers now. The best part is that it's scriptable and I've already created a better PowerShell script this time around. I will have to follow up in the morning by removing the bad PRs and starting a new PR.

**Comment by WildByDesign** [comment]
PR: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15212

Please have a look, everything should be proper now. Also, I did most of my testing while in dark mode. So please do some testing in light mode as well to ensure it is good there too.

**Comment by WildByDesign** [comment]
As per @ozone10 's great suggestions, I am upsizing the toolbar icons for much better sharpness and clarity. Particularly at the smaller sizes. Please understand that it might take me up to a week to do this. I have to grab 3 SVG's (16px, 20px, and 24px) and manually edit each one. Multiply that by 46 icons. So I have to manually edit 138 SVG files if we want this to be the absolute best. Then I can automatically create inverted copies for the other 138 needed to dark mode.

I have done the process with two icons so far. But trust me when I say that this will be worth while. The difference is night and day.

**Comment by ozone10** [comment]
@WildByDesign 
could you provide some svg files for testing, some which have disabled state like save all, cut, or undo, ...

**Comment by WildByDesign** [comment]
@WildByDesign For sure.
[examples.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/15497427/examples.zip)
You'll notice that the icons that end with _off do not have inverted SVG's. My script inverts those ones during the svg2png script.

Last night I worked on tooling so that I can fly through the icons now. I have two scripts now that take care of all converting, any inverting, packing into ICO, and renaming the icons. Previously, I had all the conversion done with scripting but I had to rename all of the icons manually at the end which was not fun. I think I love PowerShell now.

**Comment by WildByDesign** [comment]
@ozone10  Can you please test these? It's the first 3 icons on the toolbar. Please compile into your build and particularly test the size (especially smaller sizes) and also the disabled state of the Save button. Check the disabled state of the Save button in both light and dark modes. I have followed your suggestion on the disabled state colors.

[.icons.zip](https://github.com/notepad-plus-plus/notepad-plus-plus/files/15499666/default.icons.zip)

**Comment by ozone10** [comment]
@WildByDesign 
looks good, just save button seems to be smaller than other button.

**Comment by WildByDesign** [comment]
Before I submit the PR for the updated Fluent icons, I wanted to share where it's at right now. It took about a whole week to modify all of the SVGs to ensure size, clarity and most importantly the symmetry amongst all of the icons.

Everything is done, however, I do still want to fix the Word Wrap icon and possibly another icon or two. I'm hoping to submit the PR tomorrow (Thurs.) as long as all goes well.

But for today, I really wanted to show how things look right now. Please view at 100% to see at it's best.

![npp-fluent](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/26308319/d0a23455-eb74-46a8-aaf2-4f0616724ced)

**Comment by ozone10** [comment]
@WildByDesign 
Looks good, but aren't icons for folder as workspace switched?

**Comment by WildByDesign** [comment]
Yes, you are right. I did have the following briefly:

![npp-fluent-workspace](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/26308319/7e662247-fcd2-4ccb-8f78-6d2b00f9bcc9)

Do you prefer this one with the folder and toolbox?

**Comment by WildByDesign** [comment]
Alright I've got the new PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15253) submitted now. A lot of hand crafted attention was put into the symmetry between all icons.

I have compiled this into my build and tested at 100%, 125% and 150% scaling all on dark mode. I need somebody to further test more scaling scenarios in light mode as well.

I can easily update any of the icons individually as needed.

@ozone10 Have you done a commit since 8.6.7 that affects the highlighting (hot track?) of toolbar icons?

I must have missed that commit when browsing through. But what a nice surprise! It caught my attention immediately and looks fantastic. It seems to resolve the Windows 11 (blue highlighting in dark mode) issue and looks incredible. Cheers to that!

**Comment by WildByDesign** [comment]
One thing that I have wanted to tackle for a long time now is the Close button on the tabs. As far as I know, that is still BMP format and has issues with scalability and is wonky with the whole transparent background and such.

Is it possible to change the Close button on the tabs to ICO format?

I would also have more creative freedom to make it extra special. The BMPs for the Close buttons on the tabs always presented weird issues for me in the past and limited what I wanted to really do with them.

**Comment by WildByDesign** [comment]
@ozone10 
> 
> btw Search result panel does not have fluent icon ![image](https://private-user-images.githubusercontent.com/55940305/332855171-b9205ad0-2ea4-4451-9f4a-dfb2465b3859.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NjIxMjMsIm5iZiI6MTcxNzc2MTgyMywicGF0aCI6Ii81NTk0MDMwNS8zMzI4NTUxNzEtYjkyMDVhZDAtMmVhNC00NDUxLTlmNGEtZGZiMjQ2NWIzODU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA3VDEyMDM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4YzYyMmRkMmYwMzhhNTQ0ZGViODkyZGY2ODMxNTRmNzlmMzIyYzFkMTYxNWM5NDA0NmMxMTMyY2MyMDdhZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jTZr2JGjv9rVyDE86lS-j7bozII_tmafzmTr4_76knE)

Sorry for the barrage of messages, but I don't want to forget some of these details.

![npp-fluent-results](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/26308319/cde21a9c-2369-41e5-817b-eb71606d6412)

Do any of these represent the Search Results icon well? Which do you prefer?

Also, how can I see this Search Results panel in Notepad++?

I have tried all sorts of different search options and cannot figure out how to bring up that Search Results panel. Please let me know so that way I can see the icon in various scaling scenarios to ensure that I'm doing it right. Thank you.

**Comment by donho** [comment]
> Do any of these represent the Search Results icon well? Which do you prefer?

They are gorgeous! I would say the 1st ones. What do you think @ozone10 ?

Also, how can I see this Search Results panel in Notepad++?

**Step 1:**

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/e013556f-ce7d-4fb8-a28a-934b753c54cd)

**Step 2:**

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/9308c7f4-925d-4831-ab60-d0d450c3ec73)

**Comment by ozone10** [comment]
@WildByDesign 
> **Question:** Is this fix something that the LibreOffice developers could reasonably do? They still suffer from that toolbar icon blue highlighting in dark mode on Windows 11.

if they use win32 they can use custom draw

Could you check Show all characters icon, it looks strange for me.

>Close button ...

I quiet busy now, but eventually I would like to replace all BMP files. 

> Do any of these represent the Search Results icon well? Which do you prefer?

Personally I like the 3rd one.

> Also, how can I see this Search Results panel in Notepad++?

Open Find dialog and press Find All in current Document

**Comment by ozone10** [comment]
@WildByDesign 
it's almost unnoticeable but could you move icons little to the right, except for document list which could be little to left for symmetry.

**Comment by WildByDesign** [comment]
Thank you both for your quick and thorough responses, I appreciate it.

> if they use win32 they can use custom draw

I've heard that they use the Qt framework, so they may be out of luck unfortunately.

> Could you check Show all characters small icons, it looks strange for me.

I see what you mean. The Show All Characters icon is quite blurry for the 16x16 for some reason. I will fix that.

> Personally I like the 3rd one. It looks similar to current standard.

I agree with you. I think that the 3rd one represents the overall icon set.

> it's almost unnoticeable but could you move icons little to the right, except for document list which could be moved little to left for symmetry.

Yes, I see what you mean and I can fix those quite easily. I just have to shift some of the SVGs over a pixel or two. Everything after that is completely scripted and automated now. 

It was definitely worthwhile to create the scripts to export the SVGs to PNG, then stuff the PNGs into ICO files automatically named according to the Notepad++ icon filenames.

**Comment by WildByDesign** [comment]
@ozone10 I have found something interesting under 100%, 125% and 150% scaling. All of those (and likely higher scaling) all add 1 extra pixel of space on the right side of each icon,  within the square that is drawn around each icon.

Top, bottom and left do not have an added pixel of space. But in a scenarios, a single pixel of space is added on the right of each icon, but within the square.

For example, on my 125% scaling system using 20x20, the inner section of each icon box drawn is 25x24 pixels. Including the frame is 27x26.

This gives the impression that each icon is slightly off. 

Now, there were definitely some icons that legitimately needed to be moved to the right by 1 pixel. And the document list icon needed to be moved to the left by 1 pixel. However, the majority of the icons can't be moved over because they cover the entire space (left to right). So in many cases, I can't adjust them for this extra 1 pixel of space.

Do you know if this is something that can be fixed?

I just checked a build from a few weeks ago, and in that case the extra space was being added on the left. I don't understand the programming aspect of this.

**Comment by WildByDesign** [comment]
I just did some more testing regarding the issue with the box drawn around the icons to get more details to share.

20x20 pixel icon as SVG and PNG and ICO
* space referring to padding between icon and inner side of box drawn around icon

```
Space on left	2
Space on right	2

20x20 pixel icon in Notepad++

Space of left	4
Space on right	5
```

20x20 pixel icon as SVG and PNG and ICO (shift padding to account for extra pixel)
* space referring to padding between icon and inner side of box drawn around icon

```
Space on left	3
Space on right	1

20x20 pixel icon in Notepad++

Space of left	5
Space on right	4
```

So my thought process was that since we had a perfectly centered icon with 2 px space on each side evenly, I would simply shift the icon spacing so that it was moved over to account for that extra pixel of space being added to the right.

It should have worked, but we end up with an extra pixel of space on the left side now. Icons with a perfect amount of pixels in size and space cannot be aligned evenly (horizontally) at the moment. I can't even wrap my mind around the mathematics of this.

I'm really not sure how to proceed with the horizontal symmetry of the icons right now.

**Comment by WildByDesign** [comment]
Sorry for the multiple comments in a row, again.

@ozone10 @donho I have a suggestion. I propose that we merge the PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15253) as is.

Once it is merged, I can do separate (and smaller) PRs to address individual icon issues. For example, I can do a separate PR for the Show All Characters icon, Document List icon and possibly a few others to address horizontal alignment.

I just feel a lot of weight on my shoulders when I keep recreating the big PR with ~170 icons spread across 4 directories and I worry that I might make a mistake. Right now, I know that PR is proper and correct. Although as noted, I will have to follow up with one or two smaller PRs to fix a few icons.

I still haven't had time to learn Git and therefore I'm just uploading the icons through the Github web site and creating the PR that way at the moment.

What do you guys think?

If it gets merged, I can follow up with the smaller fixes today also.

**Comment by donho** [comment]
@WildByDesign 
PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15253) has been merged.
However, PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15256) is not due to the conflict.
Could you update your master, create a new branch and redo the PR please?

**Comment by WildByDesign** [comment]
Thanks @donho 
Here is the updated PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15259) to fix the Show All Characters icon.

**Comment by donho** [comment]
@WildByDesign 
Merged. Thank you!

**Comment by WildByDesign** [comment]
@donho  You're welcome.

@ozone10  I will come back to the horizontal alignment of toolbar icons once we can figure out that issue with the 1 pixel added to the right of all toolbar icons. But for now, I am working on the icons for tab bar and toolbars from docking panels and docking panels tab bar.

I ran into another weird scaling issue while working on that Search Results (findResult.ico) icon. The original icon is 16x16.

I created 16, 20, 24 and 48 pixel icons within the new ICO. Here is what I am seeing:
```xml
Scaling		Dimensions	Using
		in NPP		Icon

100		14x14		16
125		17x17		20
150		20x20		24
```

The Dimensions in NPP refers to what is visually showing within Notepad++.

For example, at 100% scaling, the 16px icon is being scaled down to 14px.
At 125% scaling, the 20px icon is being scaled down to 17px.
At 150% scaling, the 24px icon is being scaled down to 20px.

The weird thing is, the ICO contains a 20px icon yet it still scales the 24px down to 20px instead of using the 20px.

So as you can imagine, all scaling scenarios look terrible.

I will attach an example ICO to this comment so that you can test.

[findResult.zip](https://github.com/user-attachments/files/15752578/findResult.zip)

Until this is resolved, I will try working on the Function List docking panel icons to see if they are scaling correctly or not.

**Comment by ozone10** [comment]
@WildByDesign 
PR #15267 should help with loading correct icon sizes. But docking panels still require a lot of work.

**Comment by WildByDesign** [comment]
@ozone10 
> That additional 1 px IMHO is used because when you press button there is offset, without 1 px icon will be "cut".

That is interesting and I never realized that. All these years of using Notepad++ and I never noticed that the icon moved over 1 pixel to simulate the button being pressed. That makes sense. I can work with that and align the icons as best as possible to accomodate.

> PR #15267 should help with loading correct icon sizes. But docking panels still require a lot of work.

I've compiled my local build based on this PR. It has slightly changed the display size of the icons, but it seems that the sizes still quite aren't as expected.

```
Scaling		Dimensions	After
		in NPP		PR15267

100		14x14		14x14
125		17x17		18x18
150		20x20		21x21
```

**Comment by ozone10** [comment]
@WildByDesign 
I did not change icon size to not affect current looks, because Notepad++ currently use one size ico files, it still uses 14x14.

You can change size by yourself
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/f37f2d0ffeb5e75aea6d54806b06b0f0e112aa87/PowerEditor/src/WinControls/DockingWnd/DockingCont.h#L45

**Comment by WildByDesign** [comment]
I apologize for the confusion on my part. Initially, I misunderstood which icons that PR was targeting. I kept focusing on the Search Results icon and panel which was clearly not part of that PR.

Now I understand which icons and I can go ahead and complete most icons tonight which is exciting.

**Comment by WildByDesign** [comment]
> I did not change icon size to not affect current looks, because Notepad++ currently use one size ico files, it still uses 14x14.
> 
> You can change size by yourself
> 
> https://github.com/notepad-plus-plus/notepad-plus-plus/blob/f37f2d0ffeb5e75aea6d54806b06b0f0e112aa87/PowerEditor/src/WinControls/DockingWnd/DockingCont.h#L45

This is working perfectly. The icons are displaying as 16, 20, 24, etc. and looking sharp and beautiful.

I can submit a PR for this 14 to 16 base pixel change later today along with all of your the updated icons. They are all done but I am reviewing them in all scaling scenarios first before submitting the PR which should be today.

**Comment by WildByDesign** [comment]
I've just added a PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15285) for the panel tabbar and toolbar icons.

Next, I will work on the monitoring, readonly, saved and unsaved icons which should not take very long but I will do that in another PR later.

**Comment by WildByDesign** [comment]
@CennoxX  Yes, absolutely. The only problem right now is that all of those treeview icons are still in BMP format. I don't know how to do the programming that is required to switch the BMP to ICO. But once the code is there, I am ready to do those icons.

**Comment by ozone10** [comment]
@WildByDesign 
could you provide search results tab bar icons, and use filled versions. Thx.

@CennoxX 
> Are these folder/document icons of the pannel tabbar also planned?

Yes. But panels need firstly basic dpi support, then there can be work on each internal panels.

**Comment by WildByDesign** [comment]
@ozone10  Here is the latest Search Results icon (filled) in both light and dark.
[findResult.zip](https://github.com/user-attachments/files/15822165/findResult.zip)

**Comment by ozone10** [comment]
@WildByDesign 
good work, check PR #15286

**Comment by WildByDesign** [comment]
> Saved state in dark mode doesn't have icon.

True, it's basically a blank placeholder so that no icon appears. 

I'm pretty sure that we don't have any code yet to have separate light mode icons for these. But I will provide both light and dark icons and whenever you are not too busy you could add the necessary code or possibly @donho could. I should be ready to provide those later today or early tomorrow morning depending.

**Comment by WildByDesign** [comment]
> good work, check PR #15286

Thank you. Great work yourself as well. I appreciate all of your time and efforts. Also, thank you for pushing me harder on the icons because that resulted in significantly better quality icons in the end. It also got me to learn so many cool tricks and techniques in Inkscape which I am thankful for. 

Plus I've got this entire process scripted now which was crucial considering how many icons (and sizes) there are.

I will build with that PR later in the evening to test the Search Results changes.

**Comment by WildByDesign** [comment]
Tab Bar PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15287)

This is dark mode only since we never did have these for light mode. Light mode continued to use the tab bar icons from Standard. I will attach the light mode icons to this thread soon.

**Comment by donho** [comment]
PR https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15286 is merged. Thank you @WildByDesign & @ozone10  for your work.



>> Saved state in dark mode doesn't have icon.

> True, it's basically a blank placeholder so that no icon appears.

I suppose that you're talking about the save state on document tab.
But the blank (absence of icon) as saved state was your idea @WildByDesign , wasn't it?

What kind of icon you want to use instead?

**Comment by WildByDesign** [comment]
> I suppose that you're talking about the save state on document tab. But the blank (absence of icon) as saved state was your idea @WildByDesign , wasn't it?
> 
> What kind of icon you want to use instead?

Yes, that was a few years ago now. My intention was to just keep it the same and to not make any major changes. So it should be just completely transparent the same as the previous icon.


> Here is the screenshot of binary of PR #15286:

I'm not sure exactly what that PR did but that does look terrible. Try to build it without that PR. I just built from Master branch right before that PR and the icons all look proper.

**Comment by WildByDesign** [comment]
@donho  Could you try to add this icon manually overwriting the newer version of it?
[saved.zip](https://github.com/user-attachments/files/15825882/saved.zip)
 
This is the original transparent saved.ico that was used prior to that PR. I wonder if it works any better.

**Comment by WildByDesign** [comment]
> Here is the screenshot of binary of PR #15286:

I just got a chance to test with the latest master (including that PR) and I can't reproduce that visual glitch with the saved icon. Everything looks good and proper. I tried dark mode and light mode and I don't see that square at all. I'm not sure what would be causing that.

**Comment by donho** [comment]
@WildByDesign 

>>Here is the screenshot of binary of PR https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15286:

> I'm not sure exactly what that PR did but that does look terrible. Try to build it without that PR. I just built from Master branch right before that PR and the icons all look proper.

My bad, I meant Tab Bar PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15287)

> Could you try to add this icon manually overwriting the newer version of it?

It looks better:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/90293/aecbcdd6-6b72-4607-82b9-7c56f55131e9)

Could you update your PR please?

**Comment by WildByDesign** [comment]
@donho 
> Could you update your PR please?

Absolutely. I've just added another commit which reverts the `saved.ico` icon back to the original that did not have any visual glitches. It's just going through the automated tests right now so I hope that it will be successful. If it goes through testing successfully, please try and make sure that the glitch is still gone. Thank you.

**Comment by WildByDesign** [comment]
I've added a small PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15288) to improve the alignment of the Document List, Undo and Redo icons.

**Comment by WildByDesign** [comment]
@donho I have a concept for the tabbar close buttons that I wanted to do years ago. The basics of the concept is hiding the close button on the inactive tabs until hovered. That  way it keeps things clean and less distracting. Could you please share your opinion on this?

![Animation](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/26308319/4e2e7c0b-dc4f-4592-84c6-baae01c55a60)

It's an animated gif. Please disregard the silly yellow highlighting of the mouse clicks.

This is a fully functional concept. However, it would really require converting the BMP to ICO to have proper transparency because these close button BMPs on the tabbar never properly supported transparency in my testing. Plus they don't scale anyway.

**Comment by donho** [comment]
@WildByDesign 
It is clearly a feature request, and it doesn't belong here.
Could you create rather a feature request please?

**Comment by WildByDesign** [comment]
Sorry about that. I've submitted a feature request (https://github.com/notepad-plus-plus/notepad-plus-plus/issues/15298).

**Comment by WildByDesign** [comment]
I've added the PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15321) for the updated Fluent close buttons.

**Comment by WildByDesign** [comment]
@ozone10  Regarding your recent PR (https://github.com/ozone10/notepad-plus-plus/commit/5d1de3ccc41138f2b6259974c6d85fb122468e93), do we need a dark mode Fluent variant of these icons? And if Yes, what sizes do we need?

**Comment by ozone10** [comment]
@WildByDesign
base size is 16x16. I will implement logic to use fluent icons later. you can replace standard icons for test, but don't do PR yet.

**Comment by WildByDesign** [comment]
@ozone10 Here is what I have come up with for the treeview icons. Most are custom made.

![npp-treeviewicons](https://github.com/user-attachments/assets/7f17a11d-1b2e-49a1-bc2b-82ca29cb59ed)

[treeview.zip](https://github.com/user-attachments/files/16214304/treeview.zip)

Please compile into your own build and have a look. In Folder as Workspace, please have a close look at the root folder and subfolders. I had to find a way to differentiate between the two and that was what seemed to work out the best. 

Also, have a look at the Project Panel project_work_space icon as well. I had to come up with a way to indicate when the workspace was unsaved and it seemed best to me to combine the pencil icon with it similar to the pencil icon used in the tabbar.

Let me know if there are any changes that should be made. Otherwise, you can add these to your own PR whenever you are ready.

**Comment by ozone10** [comment]
I will look into it later after relevant PRs are merged.

**Comment by WildByDesign** [comment]
@ozone10 @donho Just a friendly reminder that the relevant PRs have been merged. The dark-mode/Fluent treeview icons from a couple of posts back are complete but just need code in Notepad++ to support them. The treeview.zip archive that is attached to that comment contains the up-to-date icons.

**Comment by donho** [comment]
@WildByDesign 
> The dark-mode/Fluent treeview icons from a couple of posts back are complete but just need code in Notepad++ to support them. 

Sorry, what kind of code do we need to support treeview icons?

> The treeview.zip archive that is attached to that comment contains the up-to-date icons.

I can always commit them, but it's better that you commit it by creating a PR, since it's your contribution.

**Comment by WildByDesign** [comment]
> Sorry, what kind of code do we need to support treeview icons?

I could be wrong (I'm not a programmer), but I don't think that there is code to support the switching of icon sets for the treeview icons specifically when changing between light mode and dark mode.

If I committed these icons, I believe that it would overwrite the original treeview icons that are used for light mode.

**Comment by ozone10** [comment]
@WildByDesign 
unfortunately I am currently very busy, so will be very slow.
But here is basic PR #15532.

**Comment by donho** [comment]
> But here is basic PR https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15532.

@ozone10 
Thank you for the PR.
By "basic" you mean there are somethings missing in the PR?
If yes, what they are?

**Comment by ozone10** [comment]
@donho 
treeview does not handle dpi related messages yet.

@WildByDesign 
could you create fluent treeview icons for light mode?

**Comment by WildByDesign** [comment]
> could you create fluent treeview icons for light mode?

Absolutely. Here you go:

[.treeview-light.zip](https://github.com/user-attachments/files/16580126/default.treeview-light.zip)

**Comment by ozone10** [comment]
@WildByDesign 
could you do PR, just replace icons in `icons\light\panels\treeview` and test them with light theme and non-standard toolbar icons.

**Comment by WildByDesign** [comment]
@ozone10 Yes, for sure. I won’t be able to test (and submit PR) until later in the day though but I will respond here later with how it goes.

**Comment by WildByDesign** [comment]
Here is the PR (https://github.com/notepad-plus-plus/notepad-plus-plus/pull/15541) for the light mode treeview icons.

I can't seem to test it at all, unfortunately. For the last few builds, as soon as I click on the Light mode radio button on the Dark Mode section of the settings Notepad++ freezes instantly and the CPU for that core stays at 100% until I kill the Notepad++ process.

I'm hoping that it's just something wrong with my config. I use my Notepad++ in portable mode with a config.xml file.

**Comment by donho** [comment]
@WildByDesign 
Works for me:

![image](https://github.com/user-attachments/assets/6c530370-644c-4835-987b-76e031b599a5)

**Comment by donho** [comment]
@WildByDesign 
> It was definitely my config causing the crash.

BTW, could you share your config which cause the crash here, if it's possible? The crash should not happens in anyway.
If I can reproduce it, I'll fix it.

**Comment by WildByDesign** [comment]
@donho

> BTW, could you share your config which cause the crash here, if it's possible? The crash should not happens in anyway.
> If I can reproduce it, I'll fix it.

I tracked it down a bit more and was able to reproduce it with a new config as well. It seems to be a glitch between Folder as Workspace and the Dark Mode section of Preferences.

My Notepad++ directory happens to have 56,000+ files within it for all of the icon work. I had that directory set up as Folder as Workspace. If I try to change between Light or Dark Mode or change any of those Customized colors on the Dark Mode section in Preferences, notepad++ locks up one CPU core for around 5-6 minutes at 100%.

Repro:

- download latest Fluent icons (https://github.com/microsoft/fluentui-system-icons/archive/refs/tags/1.1.253.zip)
- extract to a specific directory (zip contains some 96.000+ small files)
- set that directory as Folder as Workspace
- try to change between Dark or Light modes (or customize colors) in Dark Mode section of Preferences
- Notepad++ should lock up for 5+ minutes with constant CPU maxed out

**Comment by donho** [comment]
@WildByDesign 
Thank you for your instructions.

Here is my tests:

Test1
**Switching** - using Filled Fluent UI (small):
`v8.6.9 release`  lite->Dark: 01:23  dark->lite: 00:58
`With new icons`   lite->Dark: 01:27  dark->lite: 00:58

**Startup Notepad++** - current master's version:
`With Classic icons`: 00:07
`With new icons`: 00:07

So I can't reproduce the performance issue that you encountered - there's no issue for me for the new added icons.
However, there's still the room for improving the switching performance, since the startup takes only 7 second, no reason for me that the switching takes so long.
Note that I use a Gamer PC so the test time on another ones could be different.

**Comment by WildByDesign** [comment]
@donho 
> Thank you for your instructions.

You're welcome. I am always happy to help.

> However, there's still the room for improving the switching performance, since the startup takes only 7 second, no reason for me that the switching takes so long. Note that I use a Gamer PC so the test time on another ones could be different.

My testing was done on a laptop. Since yours was done on a Gamer PC and mine on a laptop, it seems that the performance of switching (combined with Folder as Workspace with many files), the results are quite variable depending on users computer specifications. It definitely could be improved, however, it's not a high priority issue of course.

**Comment by donho** [comment]
@WildByDesign 
An issue about the concerning performance has been create here: #15556

**Comment by xomx** [comment]
@ozone10 
Are the N++ panels (and possible controls within) going to be fully dpi-fixed in the near future?

I'm asking because my eyes are only going to get worse and this becomes more and more a problem for me (the following pic was taken on a 192 dpi screen, with vanilla v8.9):

<img width="757" height="331" alt="Image" src="https://github.com/user-attachments/assets/b450a6bd-8e3c-4a18-a023-b41fba7a84e4" />

The big pic above isn't looking so bad but what I usually see on a hi-dpi display looks like:

<img width="378" height="166" alt="Image" src="https://github.com/user-attachments/assets/d802cc5e-5680-4aef-ac18-ff945f84528d" />

It's almost unusable.

I've fixed that ListBox dpi-handling for myself but don't want to possibly add more work for you later, if I don't implement it fully right or not according to your overall N++ concept and release my CH-PRs before you update the panels. That's why I'm asking.

---

In my version, I've also fixed the ListBox items lines separation (without a custom owner draw) by increasing a little bit the default LB base 16pix items height to 18, which in turn fixed the LB item box selection border drawing itself, as now there remains the borders of the previously selected items - visible in the 1st pic above):

<img width="771" height="316" alt="Image" src="https://github.com/user-attachments/assets/0015d9c0-15f3-428d-a583-e73559d39b72" />

<img width="386" height="158" alt="Image" src="https://github.com/user-attachments/assets/fcedd450-e344-4f13-8b9f-1a85ebf082cd" />

---

(There is also a new CH-panel menu visible, I plan to add some commands to fix some CH-panel FRs and my needs later:)

<img width="865" height="363" alt="Image" src="https://github.com/user-attachments/assets/293af556-5b9d-41b0-a0e6-c84f6a32822f" />

**Comment by ozone10** [comment]
@xomx 
IIRC only Find dialog size and position, panels and plugin support is missing, and mainly implementing support for panels is really complex. You need to take into account docked state, saving positions, and how to handle plugins.

I am currently focusing on tinyxml replacement, so feel free to do PR.
